src/excited_workflow/carbon_tracker.py

```diff
@@ -0,0 +1,275 @@
+
+    for group in range(groups):
+        target_ds, prediction = groupwise_cross_validation(
+            df_group, group, x_keys, y_key
+        )
+        rmse = calculate_rmse(prediction["prediction_label"], target_ds[y_key])
+        rmse.to_netcdf(output_dir / f"rmse{group}.nc")
+        create_rmseplot(rmse)
+        plt.savefig(output_dir / f"rmseplot{group}.png")
+        plt.close()
+        create_scatterplot(prediction["prediction_label"], target_ds[y_key])
+        plt.savefig(output_dir / f"scatter{group}.png")
+        plt.close()
+        text = (
+            text
+            + f"\n ### Validation plots for group {group} \n"
+            + f"**RMSE map** \n ![image]({output_dir}/rmseplot{group}.png) \n"
+            + f"**Scatter plot** \n ![image]({output_dir}/scatter{group}.png) \n"
+        )
+
+    with open(output_dir / "model_description.md", "w") as file:
+        file.write(text)
+
+
+def save_model(ds: xr.Dataset, x_keys: list[str], y_key: str, output_dir: Path) -> None:
+    """Create lightgbm model for whole dataset and save with ONNX.
+
+    Args:
+        ds: dataset used for model creation.
+        x_keys: list of input variables.
+        y_key: target variable name.
+        output_dir: path to output directory.
+    """
+    df = ds.to_dataframe().dropna()
+
+    pycs, model = train_model(df, x_keys, y_key)
+    x_test = pycs.get_config("X_test").to_numpy()
+
+    lightgbm_onnx = onnxmltools.convert_lightgbm(
+        model, initial_types=[("X", DoubleTensorType([None, x_test.shape[1]]))]
+    )
+
+    with open(output_dir / "lightgbm.onnx", "wb") as f:
+        f.write(lightgbm_onnx.SerializeToString())
+
+
+if __name__ == "__main__":
+    client = Client()
+
+    cb_file = Path("/data/volume_2/EXCITED_prepped_data/CT2022.flux1x1-monthly.nc")
+    regions_path = Path("/data/volume_2/EXCITED_prepped_data/regions.nc")
+    output_path = Path("/home/cdonnelly")
+
+    desired_data = [
+        "biomass",
+        "spei",
+        "modis",
+        "era5_monthly",
+        "era5_land_monthly",
+        "copernicus_landcover",
+    ]
+
+    x_keys = [
+        "d2m",
+        "mslhf",
+        "msshf",
+        "ssr",
+        "str",
+        "t2m",
+        "spei",
+        "NIRv",
+        "skt",
+        "stl1",
+        "swvl1",
+        "lccs_class",
+    ]
+    y_key = "bio_flux_opt"
+
+    time = datetime.datetime.now().strftime("%Y-%m-%d_%H_%M")
+    output_dir = output_path / f"carbon_tracker-{time}"
+    output_dir.mkdir(parents=True, exist_ok=True)
+
+    ds_cb = xr.open_dataset(cb_file)
+    ds_cb = excited_workflow.utils.convert_timestamps(ds_cb)
+    ds_input = xr.merge(
+        [
+            datasets[name].load(freq="monthly", target_grid=ds_cb)
+            for name in desired_data
+        ]
+    )
+    ds_na = mask_region(regions_path, cb_file, "transcom_regions", ds_input)
+    validate_model(ds_na, 5, x_keys, y_key, output_dir)
+    save_model(ds_na, x_keys, y_key, output_dir)
```

Comment:
```suggestion
    model = train_model(ds_na, x_keys, y_key)
    save_model(model, x_keys, y_key, output_dir)
```
I still think it would be nicer to split this up this way.